packages/runtime/src/helpers/functionsMetaData.ts

```diff
@@ -34,23 +47,20 @@ export interface FunctionInfo {
  */
 export const writeFunctionConfiguration = async (functionInfo: FunctionInfo) => {
   const { functionName, functionTitle, functionsDir } = functionInfo
-  const pluginPackagePath = '.netlify/plugins/package.json'
-  const moduleRoot = resolveModuleRoot(NEXT_PLUGIN)
-  const nodeModulesPath = moduleRoot ? join(moduleRoot, 'package.json') : null
-
-  const nextPluginVersion =
-    (await getNextRuntimeVersion(nodeModulesPath, true)) ||
-    (await getNextRuntimeVersion(pluginPackagePath, false)) ||
-    // The runtime version should always be available, but if it's not, return 'unknown'
-    'unknown'
+  const pluginVersion = await nextPluginVersion()
 
   const metadata = {
     config: {
       name: functionTitle,
-      generator: `${NEXT_PLUGIN_NAME}@${nextPluginVersion}`,
+      generator: `${NEXT_PLUGIN_NAME}@${pluginVersion}`,
```

Comment:
Could this call the function you've added below instead of a string template as that already generates the generator name with the plugin version?

Reply:
This has been done, thanks Marc!